mrjob/emr.py

```diff
@@ -2454,7 +2450,14 @@ def _pool_hash(self):
 
         if self._opts['bootstrap_mrjob']:
             things_to_hash.append(mrjob.__version__)
-        return hash_object(things_to_hash)
+
+        things_json = json.dumps(things_to_hash, sort_keys=True)
+        if not isinstance(things_json, bytes):
+            things_json = things_json.encode('utf_8')
```

Comment:
Huh, really? Was trying to match [Standard Encodings](https://docs.python.org/3/library/codecs.html#standard-encodings) from the Python docs.

Yeah, was considering having a function for this, but it only happens like six times in the code base, so I decided to err on the side of readability (less indirection). Whereas `to_string()` actually *does* something, so I relented and made it a function.